flang/lib/Semantics/resolve-names.cpp

```diff
@@ -2828,6 +2828,17 @@ Scope &ScopeHandler::NonDerivedTypeScope() {
   return currScope_->IsDerivedType() ? currScope_->parent() : *currScope_;
 }
 
+static void SetImplicitCUDADevice(bool inDeviceSubprogram, Symbol &symbol) {
```

Comment:
Maybe the calls to this function should be conditional on `inDeviceSubprogram` instead of passing the condition to it as an argument.